nifi-ngsild-ckan-processors/src/main/java/egm/io/nifi/processors/ckan/ngsild/NGSIUtils.java

```diff
@@ -114,6 +70,69 @@ public NGSIEvent getEventFromFlowFile(FlowFile flowFile, final ProcessSession se
         return event;
     }
 
+    private AttributesLD parseNgsiLdAttribute(String key, JSONObject value) {
+        String attrType;
+        String attrValue="";
+        String datasetId;
+        ArrayList<AttributesLD> subAttributes=new ArrayList<>();
+
+        attrType = value.getString(NGSILD_TYPE);
+        datasetId = value.optString(NGSILD_DATASET_ID);
+        if (NGSILD_RELATIONSHIP.contentEquals(attrType)) {
+            attrValue = value.get(NGSILD_OBJECT).toString();
+        } else if (NGSILD_PROPERTY.contentEquals(attrType)) {
+            attrValue = value.get(NGSILD_VALUE).toString();
+        } else if (NGSILD_GEOPROPERTY.contentEquals(attrType)) {
+            attrValue = value.get(NGSILD_VALUE).toString();
+        }
+        Iterator<String> keysOneLevel = value.keys();
+        while (keysOneLevel.hasNext()) {
+            String keyOne = keysOneLevel.next();
+            if (IGNORED_KEYS_ON_ATTRIBUTES.contains(keyOne)){
+                // Do Nothing
+            } else if (keyOne.equals(NGSILD_OBSERVED_AT) || keyOne.equals(NGSILD_UNIT_CODE)){
+                subAttributes.add(
+                        new AttributesLD(keyOne,"NonReifiedProperty", value.getString(keyOne), "",false,null)
+                );
+            } else {
+                JSONObject value2 = value.getJSONObject(keyOne);
+                String subAttrType=value2.get(NGSILD_TYPE).toString();
+                if (NGSILD_RELATIONSHIP.contentEquals(subAttrType)){
+                    String subAttrValue = value2.get(NGSILD_OBJECT).toString();
+                    subAttributes.add(new AttributesLD(keyOne,subAttrType,subAttrValue, "",false,null));
+                }else if (NGSILD_PROPERTY.contentEquals(subAttrType)){
+                    String subAttrValue = value2.get(NGSILD_VALUE).toString();
+                    subAttributes.add(new AttributesLD(keyOne,subAttrType,subAttrValue, "",false,null));
+                }else if (NGSILD_RELATIONSHIP.contentEquals(subAttrType)){
```

Comment:
`NGSILD_RELATIONSHIP` is already handled 6 lines above. Guess it should be `NGSILD_GEOPROPERTY` here.